app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -3207,7 +3211,7 @@ class BrowserTabFragment :
             it.clearSslPreferences()
 
             it.settings.apply {
-                clientBrandHintProvider.setDefault(this)
+                withContext(dispatchers.io()) { clientBrandHintProvider.setDefault(this@apply) }
```

Comment:
is it safe to do all of `clientBrandHintProvider.setDefault` on the `io` thread now, whereas it was all `main` before?

should it be `suspend` instead, and inside you could just do the FF check on `io` leaving all the other logic to run as is? 

if you're fully confident it's a safe change, what you have is fine. but if you have any doubt or want to de-risk further, can consider that suggestion.